File: Proyecto Final/utils/permissions.py
STAFF_ROLES = {"Administrador", "Coordinador"}
ADMIN_ROLES = {"Administrador"}
VOTING_ROLES = {"Administrador", "Coordinador", "Miembro"}


def get_user_role(user: dict | None) -> str:
    if not user:
        return "Invitado"
    return (user.get("rol") or "Invitado").strip()


def has_any_role(user: dict | None, allowed_roles: set[str] | list[str] | tuple[str, ...]) -> bool:
    return get_user_role(user) in set(allowed_roles)


def can_manage_catalogs(user: dict | None) -> bool:
    return has_any_role(user, STAFF_ROLES)


def can_manage_finances(user: dict | None) -> bool:
    return has_any_role(user, STAFF_ROLES)


def can_manage_proposals(user: dict | None) -> bool:
    return has_any_role(user, STAFF_ROLES)


def can_vote(user: dict | None) -> bool:
    return has_any_role(user, VOTING_ROLES)

File: proyecto_final-main/Proyecto Final/web/src/lib/permissions.ts
import type { UserProfile } from '../types'

export function isStaff(profile: UserProfile | null) {
  return profile?.rol === 'Administrador' || profile?.rol === 'Coordinador'
}

export function isAdmin(profile: UserProfile | null) {
  return profile?.rol === 'Administrador'
}

export function canVote(profile: UserProfile | null) {
  return Boolean(profile && ['Administrador', 'Coordinador', 'Miembro'].includes(profile.rol))
}
